tanner/api.py

```diff
@@ -38,3 +100,52 @@ def __init__(self):
             for (i, val) in enumerate(query_res):
                 query_res[i] = json.loads(val)
         return query_res
+
+    async def return_session_info(self, sess_uuid, snare_uuid= None):
+        query_res = []
+        if snare_uuid:
+            snare_uuids = [snare_uuid]
+        else:
+            snare_uuids = await self.return_snares()
+
+        for snare_id in snare_uuids:
+            sessions = await self.return_snare_info(snare_id)
+            if sessions == 'Invalid SNARE UUID':
+                continue
+            for sess in sessions:
+                if sess['sess_uuid'] == sess_uuid:
+                    return sess
+
+    async def return_sessions(self, filters):
+        query_res = []
+        if 'snare_uuid' in filters:
+            snare_uuids = [filters['snare_uuid']]
+        else:
+            snare_uuids = await self.return_snares()
+
+        matching_sessions = []
+        for snare_id in snare_uuids:
+            sessions = await self.return_snare_info(snare_id)
+            for sess in sessions:
+                if sessions == 'Invalid SNARE UUID':
+                    continue
+                match_count = 0
+                if 'user-agent' in filters:
```

Comment:
So I tried making new `method` but in `apply_filter` method you need to use 4 `if` statements which is same as now. And this will make the code more lengthy.